File: LeetCode/349两个数组的交集.java
// 可以将list, set转换为流数据进行处理，从而转换为int类型的数组
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Map<Integer, Integer> map = new HashMap<>();
        Set<Integer> set = new HashSet<>();
        for (int i = 0; i < nums1.length; i++) {
            map.put(nums1[i], map.getOrDefault(nums1[i], 1));
        } 
        for (int i = 0; i < nums2.length; i++) {
            if (map.get(nums2[i]) != null) {
                set.add(nums2[i]);
            }
        }
        // int[] ans = list.stream().mapToInt(i->i).toArray();
        int[] ans = set.stream().mapToInt(i->i).toArray();
        return ans;
    }
}

File: LeetCode/349两个数组的交集.go
func intersection(nums1 []int, nums2 []int) []int {
	set := map[int]int{}
	for _, v := range nums1 {
		set[v] = 1
	}
	var ans []int
	for _, v := range nums2 {
		// if cnt, ok := set[v]; ok && cnt > 0 {
		//     ans = append(ans, v)
		//     set[v]--
		// }
		if set[v] > 0 {
			ans = append(ans, v)
			set[v]--
		}
	}
	return ans
}